lib/vanagon/project.rb

```diff
@@ -82,6 +82,32 @@ def get_files
       files.flatten.uniq
     end
 
+    # Returns a filtered out set of components only including those
+    # components necessary to build a specific component. This is a
+    # recursive function that will call itself until it gets to a
+    # component with no build requirements
+    #
+    # @param name [String] name of component to add. must be present in configdir/components and named $name.rb currently
+    def filter_component(name)
+      components_to_include = []
+      filter_component = get_component(name)
+      return nil if filter_component.nil?
+      components_to_include << filter_component
+      filter_component.build_requires.each do |build_requirement|
+        unless get_component(build_requirement).nil?
+          components_to_include += filter_component(build_requirement)
+        end
+      end unless filter_component.build_requires.nil?
+      components_to_include
+    end
+
+    def get_component(name)
+      unless @components.select { |comp| comp.name.to_s == name.to_s }.nil?
+        return @components.select { |comp| comp.name.to_s == name.to_s }.pop
+      end
+      nil
+    end
```

Comment:
So what I'm actually looking to do is return just the component, not an array consisting of only the component. I looked and couldn't find anything in ruby that allows me to select a specific element out of an array based on some criteria. (other than this)

Ideally the return value of this function should not be an array at all, so I believe we go back to using nil if the component is not in the array.